enhancements/olm/bundle-version-overrides.md

```diff
@@ -0,0 +1,486 @@
+reviewers:
+  - "@joelanford"
+  - "@rashmigottapati"
+  - "@trgeiger"
+approvers:
+  - "@joelanford"
+api-approvers:
+  - "@everettraven"
+creation-date: 2025-10-24
+last-updated: 2025-10-27
+tracking-link:
+  - "https://issues.redhat.com/browse/OCPSTRAT-2232"
+see-also:
+  - "/https://docs.google.com/document/d/1tX0fXYuflTpTal6z0TbNkrQ7SAkYIKxQj4cPjlFAh4Q/"
+  - "https://docs.google.com/document/d/14LNdxplPKB8mbfKLVvkHHsvboh4FEJOp0thzPNGgnzA/"
+  - "https://github.com/operator-framework/api/pull/454"
+  - "https://github.com/operator-framework/operator-registry/pull/1792"
+replaces: []
+superseded-by: []
+---
+
+# Bundle Version Overrides
+
+## Summary
+
+This enhancement adds an optional `release` field to the CSV specification, enabling structured release-level versioning within operator bundles. OLM will use this field when ordering multiple bundles for the same operator. 
+
+## Motivation
+
+FBC lacks a standardized mechanism for operator repackaging. Current approaches encode release information in semver build metadata (e.g., `7.10.2-opr-2+0.1676475747.p`), conflating operator version with build/release metadata. This violates [the semver spec](https://www.semver.org) and undermines FBC's reliance on semver ordering. This enhancement enables programmatic distinction between operator version and release version, supporting repackaging workflows without version changes.
+
+### User Stories
+
+* As a cluster administrator, I want OLM to prioritize bundles with higher release versions when multiple same-version bundles are available.
+* As an operator author, I want to repackage operators without changing semantic version, clearly separating operator version from build/release information.
+* As an OLM maintainer, I want structured release version tracking in bundle metadata for reliable upgrade path resolution.
+* As a legacy bundle publisher, I want existing `olm.substitutesFor` annotations with semver build metadata automatically interpreted without modifying publishing processes.
+
+### Goals
+
+* Add optional `release` field to CSV specification
+* Maintain backward compatibility with Freshmaker-style semver build metadata
+* Implement consistent version/release comparators
+* Provide catalog template for release field adoption
+* (Future) Enable OLM to prioritize bundles by release version
+
+### Non-Goals
+
+* Changing core OLM algorithms beyond adding release comparison
+* Deprecating semver build metadata approach (both coexist)
+* Breaking compatibility with existing bundles
+* Modifying SQLite catalog schema or APIs
+* Making release field mandatory
+
+## Proposal
+
+### Workflow Description
+
+Operator authors can optionally specify a `release` field alongside the `version` field in the CSV, representing release-level versioning distinct from semantic version.
+
+**Priority Ordering:** When OLM selects between multiple bundles:
+1. Semantic `version` field remains primary
+2. Higher `release` values ordered before lower values
```

Comment:
Somewhere here, we need to define the spec and comparison logic for the `release` field.

We can say "release" uses the same format as semver pre-release, but I think we should also link to the pre-release defintion and provide some examples of valid and invalid releases.

We should also describe the details of how the comparison works.